scvi/inference/posterior.py

```diff
@@ -224,62 +224,103 @@ def sample_scale_from_batch(self, n_samples, batchid=None, selection=None):
     def differential_expression_score(self, idx1, idx2, batchid1=None, batchid2=None,
                                       genes=None, n_samples=None, M_permutation=None, all_stats=True,
                                       sample_pairs=True):
-        if n_samples is None:
-            n_samples = 5000
-        if M_permutation is None:
-            M_permutation = 10000
+        """
+        Computes gene specfic Bayes factors using masks idx1 and idx2
+
+        To that purpose we sample the Posterior in the following way:
+            1. The posterior is sampled n_samples times for each subpopulation
+            2. For computation efficiency (posterior sampling is quite expensive), instead of
+            comparing element-wise the obtained samples, we can permute posterior samples.
+            Remember that computing the Bayes Factor requires sampling
+            q(z_A | x_A) and q(z_B | x_B)
+
+        :param idx1: MASK of subpopulation cells 1
+        :param idx2: MASK of subpopulation cells 2
+        :param batchid1: batch ids of cells 1
```

Comment:
We might have to expand a little bit on that. That means that each cell from batch 1 is going to be reimputed in all the different batches of the LIST batchid1. Chenling can explain to you why this is useful in multiple batches studies. This is not the batch_id of every cell in the list idx1.